apiexport/provider.go

```diff
@@ -85,6 +87,9 @@ type Options struct {
 	// against custom virtual workspaces that are not the APIExport one but share
 	// the same endpoint semantics.
 	ObjectToWatch client.Object
+
+	// Hooks are lifecycle hook, ran for each logical cluster in the provider.
+	Hooks hooks.Hooks
```

Comment:
It's not clear from the name of the variable what sort of hooks this is supposed to be. Can you rename it to something more descriptive?

Reply:
Also, does the user care what kind of an object is passed to the callbacks? From the comment alone, it seems like it's the LogicalCluster being passed, but it's actually the binding no?

Reply:
It does, as we basically represent apibindings as "logicalcluster" or as clusters.